core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -193,8 +206,18 @@ private[spark] object HighlyCompressedMapStatus {
     } else {
       0
     }
+    val hugeBlockSizes = ArrayBuffer[Tuple2[Int, Byte]]()
+    if (numNonEmptyBlocks > 0) {
+      uncompressedSizes.zipWithIndex.foreach {
+        case (size, reduceId) =>
+          if (size > 2 * avgSize) {
```

Comment:
Yes, This really makes sense. How about:
1. For the minimum size, default value could be `Int.MaxValue`.
2. For the fraction, default value could be 2.
Actually, I spent quite a while for thinking this and didn't come up with good names for these two configurations. @mridulm could you please give some advice? And could I put this in another pr ?(I think this pr is little big now.)